frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -60,33 +73,134 @@ impl From<FeasibilityError> for MinerError {
 	}
 }
 
+/// Save a given call into OCW storage.
+fn save_solution<T: Config>(call: &Call<T>) -> Result<(), MinerError> {
+	let storage = StorageValueRef::persistent(&OFFCHAIN_CACHED_CALL);
+	match storage.mutate::<_, (), _>(|_| Ok(call.clone())) {
+		Ok(Ok(_)) => Ok(()),
+		Ok(Err(_)) => Err(MinerError::FailedToStoreSolution),
+		Err(_) => unreachable!("mutation function cannot fail; qed"),
```

Comment:
I'd advise you take a `defensive-only` approach here rather than panic, specially because this is a foreign API (`mutate`) that we're not closely monitoring, it might change or sth.

Reply:
https://github.com/paritytech/substrate/pull/8290/commits/1c896bed7576b2f5c448664bfc07660cb51e4d78